### PR TITLE
XD75

### PR DESCRIPTION
**Pull request by Hybrid2288**
My XD75 keyboard creation

## Description

It's a completely new keymap so I added a keymap.c, config.h & rules.mk following the recommandations :)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xd75/keymaps/hybrid/keymap.c [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+  TD_CEDILLE,
+  TD_DOT,
+};
+
+
+
+void dance_a_accent (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 2) {
+    register_code (KC_NLCK);
+    unregister_code (KC_NLCK);
```
Comment (zvecr) [inline keyboards/xd75/keymaps/hybrid/keymap.c:51]: I wont add suggestions for all, but you can do
```suggestion
    tap_code (KC_NLCK);
```

File: keyboards/xd75/keymaps/hybrid/keymap.c [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+#define _LY2 2
+
+
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  QMKBEST = SAFE_RANGE,
+  QMKURL
+};
+
```
Comment (zvecr) [inline keyboards/xd75/keymaps/hybrid/keymap.c:30]:
```suggestion

```

File: keyboards/xd75/keymaps/hybrid/keymap.c [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+        // when keycode QMKURL is pressed
+        SEND_STRING("https://qmk.fm/" SS_TAP(X_ENTER));
+      } else {
+        // when keycode QMKURL is released
+      }
+      break;
+  }
+  return true;
+}
+
```
Comment (zvecr) [inline keyboards/xd75/keymaps/hybrid/keymap.c:446]:
```suggestion

```

File: keyboards/xd75/keymaps/hybrid/keymap.c [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+  return state;
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (zvecr) [inline keyboards/xd75/keymaps/hybrid/keymap.c:481]:
```suggestion

```

File: keyboards/xd75/keymaps/hybrid/keymap.c [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+void matrix_init_user(void) {
+  rgblight_enable();
+  rgblight_mode(1);
+  rgblight_setrgb_user_MAIN();
+}
+
+
+//Set a color based on the layer
+uint32_t layer_state_set_user(uint32_t state) {
+  switch(biton32(state)) {
```
Comment (zvecr) [inline keyboards/xd75/keymaps/hybrid/keymap.c:461]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
  switch (get_highest_layer(state)) {
```

File: keyboards/xd75/keymaps/hybrid/keymap.c [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Layer shorthand
+#define _MAIN 0
+#define _LY1 1
+#define _LY2 2
```
Comment (drashna) [inline keyboards/xd75/keymaps/hybrid/keymap.c:21]: I would prefer to see these as enums, as they will be easier to manage, long term.

```suggestion
enum layers {
    _MAIN,
    _LY1,
    _LY2,
};
```

File: keyboards/xd75/keymaps/hybrid/keymap.c (hunk 1 above)
Comment (drashna) [reply keyboards/xd75/keymaps/hybrid/keymap.c:51]: In fact this entire array could be replaced with the `send unicode` functions. 

https://docs.qmk.fm/#/feature_unicode?id=send_unicode_string

Comment (zvecr) [reply keyboards/xd75/keymaps/hybrid/keymap.c]: I can see in 7761dcbfd2988f14ffe24d8b7350c9fa9badf226 you have updated one of the instances, but the rest really need to be addressed too.

Comment (Hybrid2288) [reply keyboards/xd75/keymaps/hybrid/keymap.c]: I reviewed the code with your comments & Drashna. 
I fixed all except the tap dance functions. I cannot test the send_unicode_string right now but I will.
Thanks for the tap_code() function :)

File: keyboards/xd75/keymaps/hybrid/keymap.c (hunk 2 above)
Comment (Hybrid2288) [reply keyboards/xd75/keymaps/hybrid/keymap.c]: Deleted this part (this was a part by the template).

File: keyboards/xd75/keymaps/hybrid/keymap.c (hunk 3 above)
Comment (Hybrid2288) [reply keyboards/xd75/keymaps/hybrid/keymap.c]: Deleted this part (this was a part by the template).

File: keyboards/xd75/keymaps/hybrid/keymap.c (hunk 4 above)
Comment (Hybrid2288) [reply keyboards/xd75/keymaps/hybrid/keymap.c]: Deleted this part (this was a part by the template).

File: keyboards/xd75/keymaps/hybrid/keymap.c (hunk 5 above)
Comment (Hybrid2288) [reply keyboards/xd75/keymaps/hybrid/keymap.c]: Fixed

File: keyboards/xd75/keymaps/hybrid/keymap.c (hunk 6 above)
Comment (Hybrid2288) [reply keyboards/xd75/keymaps/hybrid/keymap.c]: Fixed.

File: keyboards/xd75/keymaps/hybrid/keymap_french.c [hunk 7]
```diff
@@ -0,0 +1,157 @@
+enum french_key {
```
Comment (zvecr) [inline keyboards/xd75/keymaps/hybrid/keymap_french.c:1]: This file doesnt seem to be referenced at all, and should be removed.

Comment (Hybrid2288) [reply keyboards/xd75/keymaps/hybrid/keymap_french.c:1]: was a deprecated thing. removed

File: keyboards/xd75/keymaps/hybrid/keymap.c [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+void dance_i_accent (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 2) {
+    tap_code (KC_NLCK);
+    register_code (KC_LALT);
+    tap_code (KC_P0);
+    tap_code (KC_P2);
+    tap_code (KC_P3);
+    tap_code (KC_P8);
+    unregister_code (KC_LALT);
+    tap_code (KC_NLCK);
```
Comment (drashna) [inline keyboards/xd75/keymaps/hybrid/keymap.c:70]: Just a heads up, you could use Unicode and the [`send_unicode_string`](https://docs.qmk.fm/#/feature_unicode?id=send_unicode_string) or [`send_unicode_hex_string`](https://docs.qmk.fm/#/feature_unicode?id=send_unicode_hex_string).

**Pull request merged**